circus/circusd.py

```diff
@@ -162,8 +162,12 @@ def main():
             pass
         finally:
             arbiter = None
-            if pidfile is not None:
-                pidfile.unlink()
+
+    # Do not delete pid file during restart loop
+    # i.e. delete it only when exiting
+    if pidfile is not None:
+        pidfile.unlink()
```

Comment:
The PID file won't be erased if an exception is raised in the loop. I think it should, as was before the change.

Reply:
Ok, I think I get it now. It will stay in `finally` block but I will also check for `restart is False`, because that what was the issue all about.